db_plugins/db/sql/__init__.py

```diff
@@ -50,30 +51,34 @@ def get_or_create(self, model=None, filter_by=None, **kwargs):
             Tuple with the instanced object and wether it was created or not
         """
         model = self._entities[0].mapper.class_ if self._entities else model
-        query = self.session.query(model) if not self._entities else self
-        instance = (
-            query.options(load_only(*filter_by.keys())).filter_by(**filter_by).first()
-        )
-        if instance:
-            return instance, False
-        else:
-            filter_by.update(kwargs)
-            instance = model(**filter_by)
+        result = self.session.query(model).filter_by(**filter_by).first()
+        created = False
+        if result is not None:
+            return result, created
+
+        try:
+            kwargs.update(filter_by)
+            instance = model(**kwargs)
             self.session.add(instance)
-            return instance, True
+            self.session.commit()
+            result = instance
+            created = True
+        except:
+            self.session.rollback()
+            result = self.session.query(model).filter_by(**filter_by).first()
+            created = False
```

Comment:
Maybe just catch the duplicate object exception (`IntegrityError`)

Reply:
changed

Reply:
from sqlalchemy.exc import IntegrityError added